CMakeLists.txt

```diff
@@ -105,7 +108,7 @@ if("${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}" VERSION_LESS "3.5")
 endif()
 
 option(WITH_CTEST "Enable ctest integration of tests" ON)
-option(DYNAMIC_LIB "Build shared libraries" OFF)
+option(DYNAMIC_LIB "Build shared libraries" ON)
```

Comment:
I am getting strange test failures when building without shared libraries.

Maybe we should just remove the option to build static libs? Or is there a situation where this is needed?

Reply:
Strange failures in `main`, or with this branch?

Reply:
This branch

Reply:
```python
_______________________________ TestMantidConversion.test_inelastic_unit_conversion ________________________________

self = <test_mantid.TestMantidConversion testMethod=test_inelastic_unit_conversion>

    def test_inelastic_unit_conversion(self):
        import mantid.simpleapi as mantid
        eventWS = self.base_event_ws
        ws_deltaE = mantid.ConvertUnits(eventWS,
                                        Target='DeltaE',
                                        EMode='Direct',
                                        EFixed=3)
        ref = mantidcompat.from_mantid(ws_deltaE)
        da = mantidcompat.from_mantid(eventWS)
        # Boost and Mantid use CODATA 2006. This test passes if we manually
        # change the implementation to use the old constants. Alternatively
        # we can correct for this by scaling L1^2 or L2^2, and this was also
        # confirmed in C++. Unfortunately only positions are accessible to
        # correct for this here, and due to precision issues with
        # dot/norm/sqrt this doesn't actually fix the test. We additionally
        # exclude low TOF region, and bump relative and absolute accepted
        # errors from 1e-8 to 1e-5.
        m_n_2006 = 1.674927211
        m_n_2018 = 1.67492749804
        e_2006 = 1.602176487
        e_2018 = 1.602176634
        scale = (m_n_2006 / m_n_2018) / (e_2006 / e_2018)
        da.coords['source-position'] *= np.sqrt(scale)
        da.coords['position'] *= np.sqrt(scale)
        low_tof = da.bins.data.coords['tof'] < 49000.0 * sc.units.us
        da.coords['incident-energy'] = 3.0 * sc.units.meV
>       da = sc.neutron.convert(da, 'tof', 'energy-transfer')
E       IndexError: map::at

tests/compat/test_mantid.py:166: IndexError
----------------------------------------------- Captured stdout call -----------------------------------------------
ConvertUnits-[Notice] ConvertUnits started
ConvertUnits-[Notice] ConvertUnits successful, Duration 0.00 seconds
____________________________________ TestMantidConversion.test_unit_conversion _____________________________________

self = <test_mantid.TestMantidConversion testMethod=test_unit_conversion>

    def test_unit_conversion(self):
        import mantid.simpleapi as mantid
        eventWS = self.base_event_ws
        ws = mantid.Rebin(eventWS, 10000, PreserveEvents=False)
        tmp = mantidcompat.convert_Workspace2D_to_data_array(ws)
        target_tof = tmp.coords['tof']
        ws = mantid.ConvertUnits(InputWorkspace=ws,
                                 Target="Wavelength",
                                 EMode="Elastic")
        converted_mantid = mantidcompat.convert_Workspace2D_to_data_array(ws)
    
        da = mantidcompat.convert_EventWorkspace_to_data_array(
            eventWS, load_pulse_times=False)
        da = sc.histogram(da, target_tof)
        d = sc.Dataset({da.name: da})
>       converted = sc.neutron.convert(d, 'tof', 'wavelength')
E       IndexError: map::at

tests/compat/test_mantid.py:129: IndexError
----------------------------------------------- Captured stdout call -----------------------------------------------
Rebin-[Notice] Rebin started
Rebin-[Notice] Rebin successful, Duration 0.00 seconds
ConvertUnits-[Notice] ConvertUnits started
ConvertUnits-[Notice] ConvertUnits successful, Duration 0.00 seconds
_______________________________________________ test_neutron_convert _______________________________________________

    def test_neutron_convert():
        d = make_dataset_with_beamline()
>       dspacing = sc.neutron.convert(d, 'tof', 'd-spacing')
E       RuntimeError: Expected <scipp.ConstView>
E         unreachable:  <scipp.VariableView>      vector_3_float64  [m]              ()  [(0.000000, 0.000000, 0.000000)]
E         position:  <scipp.VariableView>      vector_3_float64  [m]              (position)  [(1.000000, 0.000000, 0.000000), (0.000000, 1.000000, 0.000000), (0.000000, 0.000000, 1.000000), (-1.000000, 0.000000, 0.000000)]
E         tof:  <scipp.VariableView>      float64    [µs]            (tof)  [1000.000000, 1001.000000, ..., 1008.000000, 1009.000000]
E         unreachable:  <scipp.VariableView>      vector_3_float64  [m]              ()  [(0.000000, 0.000000, -10.000000)]
E        to contain source-position.

tests/neutron/test_neutron.py:39: RuntimeError
___________________________________________ test_neutron_convert_out_arg ___________________________________________

    def test_neutron_convert_out_arg():
        d = make_dataset_with_beamline()
>       dspacing = sc.neutron.convert(d, 'tof', 'd-spacing', out=d)
E       RuntimeError: Expected <scipp.ConstView>
E         tof:  <scipp.VariableView>      float64    [µs]            (tof)  [1000.000000, 1001.000000, ..., 1008.000000, 1009.000000]
E         unreachable:  <scipp.VariableView>      vector_3_float64  [m]              ()  [(0.000000, 0.000000, -10.000000)]
E         position:  <scipp.VariableView>      vector_3_float64  [m]              (position)  [(1.000000, 0.000000, 0.000000), (0.000000, 1.000000, 0.000000), (0.000000, 0.000000, 1.000000), (-1.000000, 0.000000, 0.000000)]
E         unreachable:  <scipp.VariableView>      vector_3_float64  [m]              ()  [(0.000000, 0.000000, 0.000000)]
E        to contain source-position.

tests/neutron/test_neutron.py:46: RuntimeError
______________________________________________ test_neutron_beamline _______________________________________________

    def test_neutron_beamline():
        d = make_dataset_with_beamline()
    
>       assert sc.is_equal(
            sc.neutron.source_position(d),
            sc.Variable(value=np.array([0, 0, -10]),
                        dtype=sc.dtype.vector_3_float64,
                        unit=sc.units.m))
E       RuntimeError: Expected <scipp.ConstView>
E         tof:  <scipp.VariableView>      float64    [µs]            (tof)  [1000.000000, 1001.000000, ..., 1008.000000, 1009.000000]
E         unreachable:  <scipp.VariableView>      vector_3_float64  [m]              ()  [(0.000000, 0.000000, -10.000000)]
E         position:  <scipp.VariableView>      vector_3_float64  [m]              (position)  [(1.000000, 0.000000, 0.000000), (0.000000, 1.000000, 0.000000), (0.000000, 0.000000, 1.000000), (-1.000000, 0.000000, 0.000000)]
E         unreachable:  <scipp.VariableView>      vector_3_float64  [m]              ()  [(0.000000, 0.000000, 0.000000)]
E        to contain source-position.

tests/neutron/test_neutron.py:54: RuntimeError
```

Reply:
Not strange, this is what I mention above (by linking statically, we and up with two editions of `scipp`, the second one statically linked into the `neutron` submodule. This issue will disappear once we remove `neutron` from `scipp`.

Reply:
The error you see is because we end up with two versions of a singleton / static map, so lookup fails.

Reply:
Ok